server/sonar-process-monitor/src/main/java/org/sonar/application/process/CommandFactoryImpl.java

```diff
@@ -44,20 +50,68 @@
-  public JavaCommand createEsCommand() {
-    File homeDir = settings.getProps().nonNullValueAsFile(ProcessProperties.PATH_HOME);
-    return newJavaCommand(ProcessId.ELASTICSEARCH, homeDir)
-      .addJavaOptions("-Djava.awt.headless=true")
-      .addJavaOptions(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_OPTS))
-      .addJavaOptions(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
-      .setClassName("org.sonar.search.SearchServer")
-      .addClasspath("./lib/common/*")
-      .addClasspath("./lib/search/*");
+  public EsCommand createEsCommand(AppSettings settings) {
+    File homeDir = this.settings.getProps().nonNullValueAsFile(ProcessProperties.PATH_HOME);
+    File executable = new File(homeDir, getExecutable());
+    if (!executable.exists()) {
+      throw new IllegalStateException("Cannot find elasticsearch binary");
+    }
+
+    Map<String, String> settingsMap = new EsSettings(this.settings.getProps()).build();
+
+    File logDir = new File(settingsMap.get("path.logs"));
+    File confDir = new File(settingsMap.get("path.conf"));
+    Path jvmOptionsFile = confDir.toPath().resolve("jvm.options");
+    EsCommand res = new EsCommand(ProcessId.ELASTICSEARCH)
+      .setWorkDir(executable.getParentFile().getParentFile())
+      .setExecutable(executable)
+      .setConfDir(confDir)
+      .setLog4j2Properties(buildLog4j2Properties(logDir))
+      .setArguments(this.settings.getProps().rawProperties())
+      .setClusterName(settingsMap.get("cluster.name"))
+      .setHost(settingsMap.get("network.host"))
+      .setPort(Integer.valueOf(settingsMap.get("transport.tcp.port")))
+      .addJvmOption(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_OPTS))
+      .addJvmOption(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
+      .setJvmOptionsFile(jvmOptionsFile)
+      .setEnvVariable("ES_JVM_OPTIONS", jvmOptionsFile.toString())
+      ;
+
+    settingsMap.forEach((key, value) -> res.addEsOption("-E" + key + "=" + value));
+
+    return res;
+  }
+
+  private Properties buildLog4j2Properties(File logDir) {
+    // FIXME create a Log4jHelper which shares code with LogbackHelper to build this Properties object + not make LogbackHelper.resolveLevel public + provide correct log format, rolling policy, ...
+    String logLevel = LogbackHelper.resolveLevel(settings.getProps(), "sonar.log.level", "sonar.log.level.es").toString();
+    Properties log4j2Properties = new Properties();
+    log4j2Properties.put("status", "error");
+    log4j2Properties.put("appender.rolling.type", "RollingFile");
+    log4j2Properties.put("appender.rolling.name", "rolling");
+    log4j2Properties.put("appender.rolling.fileName", new File(logDir, "es.log").getAbsolutePath());
+    log4j2Properties.put("appender.rolling.layout.type", "PatternLayout");
+    log4j2Properties.put("appender.rolling.layout.pattern", "[%d{ISO8601}][%-5p][%-25c{1.}] %marker%.-10000m%n");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 197 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)